src/clientmap.cpp

```diff
@@ -162,7 +162,7 @@ static bool isOccluded(Map *map, v3s16 p0, v3s16 p1, float step, float stepfac,
 			is_transparent = (f.solidness != 2);
 		if(!is_transparent){
 			count++;
-			if(count >= needed_count)
+			if(count > needed_count)
 				return true;
```

Comment:
I think doing this is better.

if (count == needed_count)
     return true;
count++;

Count cannot be greater than needed count because if count == needed_count we return.